tasks/process_flakes.py

```diff
@@ -71,28 +87,27 @@ def run_impl(
                                     upserted_flake.reduced_error_id,
                                 )
                             ] = upserted_flake
+                upload.state = "flake_processed"
```

Comment:
being hit by this problem myself, I’m a bit surprised that `upload.state` is not an enum that rejects new variants. unless `flake_processed` already exists as one of those variants?

Reply:
upload.state is a char field so there's no enforcement whatsoever.

i think this field could be arbitrary since we're sharing the concept of "upload" for all three products that use them, perhaps we should reorganize the Upload to only contain information related to the storage of the raw uploaded file and keep track of the processing state of each type of upload in dedicated tables for each type of upload with enums and enforcement? 

for now i'd just like to arbitrarily set the state and encode the logic to handle each state in the app and clean up this state stuff later